src/coreos.osbuild.s390x.mpp.yaml

```
@@ -0,0 +1,653 @@
+            references:
+              - name:raw-image
+        options:
+          paths:
+            - from: input://tree/disk.img
+              to: tree:///disk.img
+      - type: org.osbuild.kernel-cmdline.bls-append
+        options:
+          bootpath: mount:///
+          kernel_opts:
+            - ignition.platform.id=metal
+        devices:
+          boot:
+            type: org.osbuild.loopback
+            options:
+              filename: disk.img
+              start:
+                mpp-format-int: '{image.layout[''boot''].start}'
+              size:
+                mpp-format-int: '{image.layout[''boot''].size}'
+        mounts:
+          - name: boot
+            type: org.osbuild.ext4
+            source: boot
+            target: /
+      # Must run after setting ignition.platform.id above
+      - type: org.osbuild.zipl.inst
+        options:
+          kernel: "1"
+          kernel_opts_append:
+            - ignition.firstboot
+          location:
+            mpp-format-int: '{image.layout[''boot''].start}'
+        devices:
+          disk:
+            type: org.osbuild.loopback
+            options:
+              filename: disk.img
+              partscan: true
+        mounts:
+          - name: root
+            type: org.osbuild.xfs
+            source: disk
+            partition:
+              mpp-format-int: '{image.layout[''root''].partnum}'
+            target: /
+          - name: boot
+            type: org.osbuild.ext4
+            source: disk
+            partition:
+              mpp-format-int: '{image.layout[''boot''].partnum}'
+            target: /boot
+  - name: raw-metal4k-image
+    stages:
+      - type: org.osbuild.copy
+        inputs:
+          tree:
+            type: org.osbuild.tree
+            origin: org.osbuild.pipeline
+            references:
+              - name:raw-4k-image
+        options:
+          paths:
+            - from: input://tree/disk.img
+              to: tree:///disk.img
+      - type: org.osbuild.kernel-cmdline.bls-append
+        options:
+          bootpath: mount:///
+          kernel_opts:
+            - ignition.platform.id=metal
+        devices:
+          boot:
+            type: org.osbuild.loopback
+            options:
+              filename: disk.img
+              start:
+                mpp-format-int: '{image4k.layout[''boot''].start}'
+              size:
+                mpp-format-int: '{image4k.layout[''boot''].size}'
+              sector-size:
+                  mpp-format-int: "{four_k_sector_size}"
+        mounts:
+          - name: boot
+            type: org.osbuild.ext4
+            source: boot
+            target: /
+      # Must run after setting ignition.platform.id above
+      - type: org.osbuild.zipl.inst
+        options:
+          kernel: "1"
+          kernel_opts_append:
+            - ignition.firstboot
+          location:
+            mpp-format-int: '{image4k.layout[''boot''].start}'
```

@nikita-dubrovskii should we be setting blocksize here too? It's an option to the `zipl.inst` stage. I ran tests and they passed without it, though ???

```suggestion
            mpp-format-int: '{image4k.layout[''boot''].start}'
          blocksize:
            mpp-format-int: "{four_k_sector_size}"
```

---

I cannot say for sure, but assume we can ignore it:
- as i remember `blocksize` is used to calculate the `offset`, but here we pass it manually (on regular disks offset is `2048`, on 4k - `256`, and both match partition `start`)
- we don't have `4k.qcow2`:
    -  `4k` installation (mostly under zVM) calls `zipl` anyhow within guest
    -  if `config.ign` contains custom 4k partitions, final step is again `zipl`